pkg/controller/ovn_dnat.go

```diff
@@ -541,3 +573,34 @@ func (c *Controller) handleDelOvnDnatFinalizer(cachedDnat *kubeovnv1.OvnDnatRule
 	}
 	return nil
 }
+
+func (c *Controller) AddDnatRule(vpcName, dnatName, externalIp, internalIp, externalPort, internalPort, protocol string) error {
+	vip := net.JoinHostPort(externalIp, externalPort)
+	backendIp := net.JoinHostPort(internalIp, internalPort)
+
+	if err := c.ovnLegacyClient.CreateLoadBalancerRule(dnatName, vip, backendIp, protocol); err != nil {
+		klog.Errorf("failed to create loadBalancer rule, %v", err)
+		return err
+	}
+
+	if err := c.ovnLegacyClient.AddLoadBalancerToLogicalRouter(dnatName, vpcName); err != nil {
+		klog.Errorf("failed to add loadBalancer to vpcName, %v", err)
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) DelDnatRule(vpcName, dnatName, externalIp, externalPort string) error {
+	vip := net.JoinHostPort(externalIp, externalPort)
+
+	if err := c.ovnLegacyClient.DeleteLoadBalancerVip(vip, dnatName); err != nil {
+		klog.Errorf("failed to delete loadBalancer rule, %v", err)
+		return err
+	}
+
+	if err := c.ovnLegacyClient.RemoveLoadBalancerFromLogicalRouter(dnatName, vpcName); err != nil {
+		klog.Errorf("failed to remove loadBalancer from vpcName, %v", err)
```

Comment:
failed to remove loadBalancer from vpcName

failed to remove loadBalancer by vpc name